AutomatticTracks/src/main/java/com/automattic/android/tracks/TracksClient.java

```diff
@@ -267,6 +274,8 @@ public void run() {
                                         mInsertEventsQueue.notifyAll();
                                     }
                                 }
+                            } else {
+                                lastNetworkErrorTimestamp = 0L;
```

Comment:
> If I got it right, the problem you've mentioned could happen if the app sends 10+ events to the library

That's exactly it.

> mNetworkQueue usually contains 1 element only, with all the events that should be sent over the wire with a single REST call.

Because of this I think we're safe. Even if it was the case that many elements are in the queue this code-path could only hit if the network latency was somehow faster than a simple thread-wake. So...it would likely never occur. Looks like it's good for a merge :+1: